src/fft.rg

```diff
@@ -279,48 +589,133 @@ function fft.generate_fft_interface(itype, dtype)
+      --R2C double to complex64
+      elseif dtype_size == 16 and real_flag then
+        format.println("Calling cufftExecD2Z (Double to Complex64)...")
+        ok = cufft_c.cufftExecD2Z(p.cufft_p, [&cufft_c.cufftDoubleReal](input_base), [&cufft_c.cufftDoubleComplex](output_base))
+      --C2C complex64 to complex64
+      elseif dtype_size == 16 then
+        format.println("Calling cufftExecZ2Z (Complex64 to Complex64)...")
+        ok = cufft_c.cufftExecZ2Z(p.cufft_p, [&cufft_c.cufftDoubleComplex](input_base), [&cufft_c.cufftDoubleComplex](output_base), cufft_c.CUFFT_FORWARD)
+      end
+
+      --Check return values of Exec
+      if ok == cufft_c.CUFFT_INVALID_VALUE then
+          format.println("Invalid value in cufftExecZ2Z")
+      elseif ok == cufft_c.CUFFT_INVALID_PLAN then
+          format.println("Invalid plan passed to cufftExecZ2Z")
+      end
+
+      --format.println("cufftExecZ2Z returned {}", ok)
+      regentlib.assert(ok == cufft_c.CUFFT_SUCCESS, "cufftExecZ2Z failed")
+      format.println("cufftExecZ2Z successful")
+
+    --Otherwise, we are in CPU mode: use FFTW if no GPU
     else
       c.printf("execute plan via FFTW\n")
-      fftw_c.fftw_execute_dft(p.p, [&fftw_c.fftw_complex](input_base), [&fftw_c.fftw_complex](output_base))
+      --R2C float to complex32: Not supported 
+      if dtype_size == 8 and real_flag then
+          format.println("Executing FFTW R2C Float to Complex32: Not Supported")
+          --fftw_c.fftwf_execute_dft_r2c(p.float_p, [&float](input_base), [&fftw_c.fftwf_complex](output_base))     --void fftw_execute_dft(const fftw_plan p, fftw_complex *in, fftw_complex *out))
+      --C2C complex32 to complex32: Not supported 
+      elseif dtype_size == 8 then
+          format.println("Executing FFTW C2C Complex32 to Complex32: Not Supported")
+          --fftw_c.fftwf_execute_dft(p.float_p, [&fftw_c.fftwf_complex](input_base), [&fftw_c.fftwf_complex](output_base))     --void fftw_execute_dft(const fftw_plan p, fftw_complex *in, fftw_complex *out))
+      --R2C double to complex64
+      elseif dtype_size == 16 and real_flag then
+          format.println("Executing FFTW R2C double to complex64")
+          fftw_c.fftw_execute_dft_r2c(p.p, [&double](input_base), [&fftw_c.fftw_complex](output_base))     --void fftw_execute_dft(const fftw_plan p, fftw_complex *in, fftw_complex *out))
+      --C2C complex64 to complex64
+      elseif dtype_size == 16 then
+          format.println("Executing FFTW C2C complex64 to complex64")
+          fftw_c.fftw_execute_dft(p.p, [&fftw_c.fftw_complex](input_base), [&fftw_c.fftw_complex](output_base))   --void fftw_execute_dft(const fftw_plan p, fftw_complex *in, fftw_complex *out))
+      end 
     end
   end
 
+
+  --execute_plan tasks are demand_(__inline tasks). This means that if the user wants it to execute it in a separate task, they must wrap the task themselves -- this is what execute_plan_task does.
```

Comment:
Let's write some better documentation for this.

Reply:
To clarify with Elliott